# 한국 토지관리 데이터 자료 분석
이 데이터에는 장소의 값과 건축면적과 층 수, 연면적, 높이 등의 데이터가 있습니다.
이 데이터에 속성 값이 없으니깐 제대로 추가해봐야겠습니다.

In [31]:
import pandas as pd

building_management_register = pd.read_csv('data/building_data.tsv', sep='|', encoding='cp949')

# 데이터를 보니깐 마지막 줄이 하나 더 만들어져있다.
building_management_register = building_management_register.drop(building_management_register.columns[-1], axis=1)

building_management_register

/var/folders/zz/y6gcr8t54cn5mw40rcb6nq3c0000gn/T/ipykernel_19049/400793865.py:3: DtypeWarning: Columns (12,13,32,35) have mixed types. Specify dtype option on import or set low_memory=False.
  building_management_register = pd.read_csv('data/building_data.tsv', sep='|', encoding='cp949')


42110-13438  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트      101동  \
0          42110-13439  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트       상가동   
1          42110-13440  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트      108동   
2          42110-13441  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트      107동   
3          42110-13442  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트  유치원,생활시설   
4          42110-13443  2  집합   강원도 춘천시 우두동 1068-1번지      동부아파트      106동   
...                ... ..  ..                    ...        ...       ...   
55873       41410-6395  2  집합     경기도 군포시 산본동 1092번지      장미아파트      상가A동   
55874       41410-6452  2  집합   경기도 군포시 산본동 1119-3번지  백두(극동)아파트        상가   
55875       41410-6456  2  집합   경기도 군포시 산본동 1119-4번지    동성백두아파트       상가동   
55876      11440-27715  2  집합    서울특별시 마포구 망원동 503번지      현진에버빌       NaN   
55877  48250-102028578  2  집합  경상남도 김해시 율하동 1334-4번지        NaN       NaN   

       42110  11800  0  1068  ...    12394    41  1.1    동   01  정기점검  \
0      42110  11800  0  1068  ...   1696.0  11.0  1.0    동  1.0  정기점검   
1      42110  11800  0  1068  ...  11495.0  41.0  1.0    동  1.0  정기점검   
2      42110  11800  0  1068  ...   8626.0  39.0  1.0    동  1.0  정기점검   
3      42110  11800  0  1068  ...    873.0   0.0  1.0    동  1.0  정기점검   
4      42110  11800  0  1068  ...   6042.0  36.0  1.0    동  1.0  정기점검   
...      ...    ... ..   ...  ...      ...   ...  ...  ...  ...   ...   
55873  41410  10400  0  1092  ...    325.0   6.0  NaN  NaN  1.0  정기점검   
55874  41410  10400  0  1119  ...   1497.0  12.0  NaN  NaN  1.0  정기점검   
55875  41410  10400  0  1119  ...    692.0   0.0  NaN  NaN  1.0  정기점검   
55876  11440  12300  0   503  ...   7491.0  40.0  NaN  NaN  1.0  정기점검   
55877  48250  13100  0  1334  ...  10181.0  32.0  NaN  NaN  1.0  정기점검   

         19941129 Unnamed: 36  Unnamed: 37    20150119  
0        19941129         NaN          NaN  20150119.0  
1        19941129         NaN          NaN  20150119.0  
2        19941129         NaN          NaN  20150119.0  
3        19941125         NaN          NaN  20150119.0  
4        19941129         NaN          NaN  20150119.0  
...           ...         ...          ...         ...  
55873  19930706.0         NaN   20190412.0  20150119.0  
55874  19930916.0         NaN   20190412.0  20150119.0  
55875  19930327.0         NaN   20190412.0  20150119.0  
55876  20031128.0         NaN   20190705.0  20151128.0  
55877  20091211.0         NaN   20190717.0  20231211.0  

[55878 rows x 39 columns]

위 데이터를 보면 비어있는 값이 많고, 또한 header가 없어서 어렵다.
우선은 header를 추가해보자.


In [32]:
building_management_register.columns = ["관리건출물대장", "대장구분코드", "대장구분코드명", "대지위치", "건물명", "동명", "시군구코드", "법정동코드", "대지구분코드", "번", "지", "특수지명", "블록", "로트", "주용도코드", "주용도코드명", "기타용도코드",
                                "기타용도코드명", "구조코드", "구조코드명", "기타구조", "세대수", "호수", "가구수", "아파트호수", "지상층수", "지하층수", "건축면적", "동연면적", "연면적", "높이", "동호구분코드", "동호구분코드명", "통보구분코드",
                                "통보구분코드명", "사용승인일", "우수건축물지정기준일", "선정일", "기준일"]

building_management_register

관리건출물대장  대장구분코드 대장구분코드명                   대지위치        건물명  \
0          42110-13439       2      집합   강원도 춘천시 우두동 1068-1번지      동부아파트   
1          42110-13440       2      집합   강원도 춘천시 우두동 1068-1번지      동부아파트   
2          42110-13441       2      집합   강원도 춘천시 우두동 1068-1번지      동부아파트   
3          42110-13442       2      집합   강원도 춘천시 우두동 1068-1번지      동부아파트   
4          42110-13443       2      집합   강원도 춘천시 우두동 1068-1번지      동부아파트   
...                ...     ...     ...                    ...        ...   
55873       41410-6395       2      집합     경기도 군포시 산본동 1092번지      장미아파트   
55874       41410-6452       2      집합   경기도 군포시 산본동 1119-3번지  백두(극동)아파트   
55875       41410-6456       2      집합   경기도 군포시 산본동 1119-4번지    동성백두아파트   
55876      11440-27715       2      집합    서울특별시 마포구 망원동 503번지      현진에버빌   
55877  48250-102028578       2      집합  경상남도 김해시 율하동 1334-4번지        NaN   

             동명  시군구코드  법정동코드  대지구분코드     번  ...      연면적    높이 동호구분코드  \
0           상가동  42110  11800       0  1068  ...   1696.0  11.0    1.0   
1          108동  42110  11800       0  1068  ...  11495.0  41.0    1.0   
2          107동  42110  11800       0  1068  ...   8626.0  39.0    1.0   
3      유치원,생활시설  42110  11800       0  1068  ...    873.0   0.0    1.0   
4          106동  42110  11800       0  1068  ...   6042.0  36.0    1.0   
...         ...    ...    ...     ...   ...  ...      ...   ...    ...   
55873      상가A동  41410  10400       0  1092  ...    325.0   6.0    NaN   
55874        상가  41410  10400       0  1119  ...   1497.0  12.0    NaN   
55875       상가동  41410  10400       0  1119  ...    692.0   0.0    NaN   
55876       NaN  11440  12300       0   503  ...   7491.0  40.0    NaN   
55877       NaN  48250  13100       0  1334  ...  10181.0  32.0    NaN   

      동호구분코드명 통보구분코드 통보구분코드명       사용승인일 우수건축물지정기준일         선정일         기준일  
0           동    1.0    정기점검    19941129        NaN         NaN  20150119.0  
1           동    1.0    정기점검    19941129        NaN         NaN  20150119.0  
2           동    1.0    정기점검    19941129        NaN         NaN  20150119.0  
3           동    1.0    정기점검    19941125        NaN         NaN  20150119.0  
4           동    1.0    정기점검    19941129        NaN         NaN  20150119.0  
...       ...    ...     ...         ...        ...         ...         ...  
55873     NaN    1.0    정기점검  19930706.0        NaN  20190412.0  20150119.0  
55874     NaN    1.0    정기점검  19930916.0        NaN  20190412.0  20150119.0  
55875     NaN    1.0    정기점검  19930327.0        NaN  20190412.0  20150119.0  
55876     NaN    1.0    정기점검  20031128.0        NaN  20190705.0  20151128.0  
55877     NaN    1.0    정기점검  20091211.0        NaN  20190717.0  20231211.0  

[55878 rows x 39 columns]

데이터 분석하면 아래와 같다.
대지 위치가 55878개 이다.
간단하게 큰 타워들 위주인거 같다.
만약에 없으면 어떻게 해야할지 모르겠다.

필요없는 데이터를 없애고 필요할거 같은거만 가져와보자.

In [34]:
building_management_register = building_management_register.loc[:, ['대지위치', '건물명', '동명', '번', '지', '주용도코드명', '기타용도코드명', '세대수', '호수', '가구수', '지상층수', '지하층수', '건축면적', '동연면적', '연면적', '높이']]

building_management_register

대지위치        건물명        동명     번  지     주용도코드명  \
0       강원도 춘천시 우두동 1068-1번지      동부아파트       상가동  1068  1       공동주택   
1       강원도 춘천시 우두동 1068-1번지      동부아파트      108동  1068  1       공동주택   
2       강원도 춘천시 우두동 1068-1번지      동부아파트      107동  1068  1       공동주택   
3       강원도 춘천시 우두동 1068-1번지      동부아파트  유치원,생활시설  1068  1  제2종근린생활시설   
4       강원도 춘천시 우두동 1068-1번지      동부아파트      106동  1068  1       공동주택   
...                      ...        ...       ...   ... ..        ...   
55873     경기도 군포시 산본동 1092번지      장미아파트      상가A동  1092  0  제2종근린생활시설   
55874   경기도 군포시 산본동 1119-3번지  백두(극동)아파트        상가  1119  3  제1종근린생활시설   
55875   경기도 군포시 산본동 1119-4번지    동성백두아파트       상가동  1119  4  제1종근린생활시설   
55876    서울특별시 마포구 망원동 503번지      현진에버빌       NaN   503  0       공동주택   
55877  경상남도 김해시 율하동 1334-4번지        NaN       NaN  1334  4  제2종근린생활시설   

           기타용도코드명    세대수    호수  가구수  지상층수  지하층수    건축면적     동연면적      연면적  \
0              아파트    0.0  35.0  0.0   3.0   1.0   432.0      0.0   1696.0   
1              아파트  150.0   0.0  0.0  15.0   1.0   804.0      0.0  11495.0   
2              아파트  112.0   0.0  0.0  14.0   1.0   643.0      0.0   8626.0   
3             체육도장    0.0   0.0  0.0   4.0   1.0   162.0      0.0    873.0   
4              아파트   78.0   0.0  0.0  13.0   1.0   483.0      0.0   6042.0   
...            ...    ...   ...  ...   ...   ...     ...      ...      ...   
55873  기타제2종근린생활시설    0.0   7.0  7.0   1.0   0.0   334.0      0.0    325.0   
55874       생활편익시설    0.0  35.0  0.0   2.0   1.0  4302.0      0.0   1497.0   
55875          소매점    0.0  16.0  0.0   2.0   1.0   231.0      0.0    692.0   
55876          아파트   56.0   0.0  0.0  15.0   1.0   519.0      0.0   7491.0   
55877          소매점    0.0   0.0  0.0   8.0   2.0   967.0  10181.0  10181.0   

         높이  
0      11.0  
1      41.0  
2      39.0  
3       0.0  
4      36.0  
...     ...  
55873   6.0  
55874  12.0  
55875   0.0  
55876  40.0  
55877  32.0  

[55878 rows x 16 columns]

## 데이터 위치별 평수 구분
건물의 평수가 문제가 있는 부분이 있다.
정말 큰 가게도 있겠지만, 상가 건물 같은 경우 건물 크기로 받아버리기 때문에 
